source/celview.cpp

```diff
@@ -136,73 +138,149 @@ void CelView::framePixelClicked(unsigned x, unsigned y)
 
 void CelView::insertImageFiles(IMAGE_FILE_MODE mode, const QStringList &imagefilePaths, bool append)
 {
-    int prevFrameCount = this->gfx->getFrameCount();
-
+    // FIXME: remove that boolean variable and separate functions so there are both append and
+    // insert ones
     if (append) {
         // append the frame(s)
         for (int i = 0; i < imagefilePaths.count(); i++) {
-            this->insertFrame(mode, this->gfx->getFrameCount(), imagefilePaths[i]);
+            this->sendAddFrameCmd(mode, this->gfx->getFrameCount(), imagefilePaths[i]);
         }
-        int deltaFrameCount = this->gfx->getFrameCount() - prevFrameCount;
-        if (deltaFrameCount == 0) {
-            return; // no new frame -> done
-        }
-        // jump to the first appended frame
-        this->currentFrameIndex = prevFrameCount;
-        this->updateGroupIndex();
     } else {
         // insert the frame(s)
+        int nextIdx = 0;
         for (int i = imagefilePaths.count() - 1; i >= 0; i--) {
```

Comment:
I'm not sure if I understand what's going on here. The original code simply inserted frames at `currentFrameIndex` in reverse order, presumably causing each one to be inserted in front of the previous one that was inserted. This code seems to use `nextIdx` to increment the index at which they get inserted so each one gets inserted behind the previous one that was inserted.

I guess because you are now adding each image separately to the undo stack, you decided to insert them in the order they appear in `imagefilePaths`? But the loop still runs backwards so aren't the frames being inserted in reverse order?

Reply:
Right, I've deleted it - as we've spoke - those were probably some remainings from my last force pushes, deleted those lines.